ui/dashboard/projects/utask-lib/src/lib/@routes/task/task.component.ts

```diff
@@ -30,19 +30,124 @@ export class TaskComponent implements OnInit, OnDestroy {
   private _task$ = new BehaviorSubject<Task | null>(null);
   readonly task$ = this._task$.asObservable();
 
-  readonly template$ = this.task$.pipe(switchMap(task => this.api.template.get(task.template_name)));
+  readonly template$ = this.task$.pipe(switchMap(task => {
+    if (!task) {
+      return of(null);
+    }
+    return this.api.template.get(task.template_name)
+  }));
 
   private _resolution$ = new BehaviorSubject<Resolution | null>(null);
   readonly resolution$ = this._resolution$.asObservable();
 
-  readonly startOver$ = combineLatest([this.meta$, this.template$, this.resolution$]).pipe(map(([meta, template, resolution]) => {
+  private _isResolver$ = combineLatest([
+    this.meta$, this.template$, this.task$, this.resolution$
+  ]).pipe(map(([meta, template, task, resolution]) => {
+    if (meta?.user_is_admin) {
+      return true;
+    }
+
+    // check if the current user is declared as resolver in the task template
+    if (meta && (template?.allowed_resolver_usernames ?? []).includes(meta.username)) {
+      return true;
+    }
+
+    // check if the current user has at least one group declared as resolver in the task template
+    if (meta && (template?.allowed_resolver_groups ?? []).some(v => meta.user_groups.includes(v))) {
+      return true;
+    }
+
+    // check if the current user is declared as resolver in the task
+    if (meta && (task?.resolver_usernames ?? []).includes(meta.username)) {
+      return true;
+    }
+
+    // check if the current user has at least one group declared as resolver in the task
+    if (meta && (task?.resolver_groups ?? []).some(v => meta.user_groups.includes(v))) {
+      return true;
+    }
+
+    // check if the current user the resolution resolver
+    if (meta && resolution && meta.username === resolution.resolver_username) {
+      return true;
+    }
+
+    return false;
+  }))
+
+  readonly canStartOver$ = combineLatest([
+    this.meta$, this.template$, this.resolution$, this._isResolver$
+  ]).pipe(map(([meta, template, resolution, isResolver]) => {
     if (['PAUSED', 'CANCELLED', 'BLOCKED_BADREQUEST'].indexOf(resolution?.state) === -1) {
```

Comment:
Any reason tokeep using `indexOf` here instead of `includes` like in the others functions ?